### PR TITLE
better sprite handling

### PR DESCRIPTION
**Pull request by Bod9001**
if it can't pass a Sprite change it saves it for later, also simplified the de-serialisation side
fixed a bug where floating animation tried to play when character wasn't set, 
stop clients from receiving character data when it does not need it
fixes surgery
fixes laying down or triggering properly

### CODE REVIEW

File: UnityProject/Assets/Scripts/Messages/Server/SpritesMessages/SpriteUpdateMessage.cs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Text;
+using Google.Protobuf.WellKnownTypes;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Messages/Server/SpritesMessages/SpriteUpdateMessage.cs]: owo wats this

File: UnityProject/Assets/Scripts/Player/PlayerEffectsManager.cs [hunk 2]
```diff
@@ -22,6 +22,7 @@ private void Awake()
 
     private void Update()
     {
+	    if (PlayerManager.PlayerScript?.PlayerSync == null ) return;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Player/PlayerEffectsManager.cs]:
```suggestion
	    if (PlayerManager.PlayerScript.OrNull()?.PlayerSync == null ) return;
```
PlayerScript is a ManagedNetworkBehaviour which I assume inherits from UnityEngine.Object

File: UnityProject/Assets/Scripts/Messages/Server/SpritesMessages/SpriteUpdateMessage.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Messages/Server/SpritesMessages/SpriteUpdateMessage.cs]: I think it got confused with the different type rider did

File: UnityProject/Assets/Scripts/Messages/Server/SpritesMessages/SpriteUpdateMessage.cs [hunk 3]
```diff
@@ -36,193 +40,156 @@ public override void Process(NetMessage msg)
...
 			{
+
+				ProcessingData.Clear();
+				ProcessingData.AddRange(UnprocessedData);
+				UnprocessedData.Clear();
+				foreach (var UnprocessedDataSingle in UnprocessedData)
+				{
+					ProcessSection(0, 0, UnprocessedDataSingle);
+				}
+
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Messages/Server/SpritesMessages/SpriteUpdateMessage.cs]: Clearing it then looping through???

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Messages/Server/SpritesMessages/SpriteUpdateMessage.cs]: oh yeah, good catch xD

**Pull request merged**